src/vmm/src/persist.rs

```diff
@@ -247,6 +251,38 @@ fn snapshot_memory_to_file(
     }
 }
 
+/// Validates that snapshot CPU vendor matches the host CPU vendor.
+#[cfg(target_arch = "x86_64")]
+pub fn validate_cpu_vendor(
+    microvm_state: &MicrovmState,
+) -> std::result::Result<(), LoadSnapshotError> {
+    let host_vendor_id = get_vendor_id().map_err(|_| {
+        LoadSnapshotError::CpuVendorMismatch("Failed to read vendor from CPUID.".to_owned())
+    })?;
+
+    // Search for vendor id entry in state CPUID.
+    for entry in microvm_state.vcpu_states[0].cpuid.as_slice().iter() {
```

Comment:
Why do we need to duplicate the information. All the cpuid entries are needed to restore the state in KVM.

Reply:
My thinking was to avoid cpuid processing in vmm and to avoid as much as possible architecture specific code in vmm, but I guess we can leave it at that. 
However, we are still duplicating code even with your approach since the logic used to get the vendor is the same with get_vendor_id in cpuid crate, right?

The 100% non-duplication approach would be to modify the `get_vendor_id` function in cpuid to this:

```
pub fn get_vendor_id(cpuid: Option<CpuidResult>) -> Result<[u8; 12], Error> {
    let vendor_entry = cpuid.unwrap_or(get_cpuid(0, 0)?);
    let bytes: [u8; 12] = unsafe {
        std::mem::transmute([vendor_entry.ebx, vendor_entry.edx, vendor_entry.ecx])
    };
    Ok(bytes)
}
```

Reply:
I am using the get_vendor_id() from the cpuid crate. I'm just doing transmute on the cpuid entry to be able to compare to the one got from get_vendor_id() and also print in logs.

Reply:
I think the good point raised here is that the cpuid-related logic could be contained in the `cpuid` crate somehow.

Aka add some function in `cpuid` crate that hides/contains the hw specific logic and just call that from here.

Reply:
I've synced with @dianpopa offline and I finally understood her point and made the code changes. PTAL.